lib/y2_module_basetest.pm

```diff
@@ -31,8 +31,23 @@ our @EXPORT = qw(is_network_manager_default
   continue_info_network_manager_default
   accept_warning_network_manager_default
   workaround_suppress_lvm_warnings
+  with_yast_env_variables
 );
 
+=head2 with_yast_env_variables
+
+ with_yast_env_variables([extra_vars]);
+
+Set environment variables for yast application.
+C<extra_vars> extends the variables that can be used. C<extra_vars> expects a string.
+ex: with_yast_env_variables("foo=bar");
+
+=cut
+sub with_yast_env_variables {
+    my ($self, $extra_vars) = shift // '';
```

Comment:
I have confused you, sorry for that. 
If you are *shift*ing, it means that you will get a scalar/element only. Thus `$extra_vars` will never be assigned and it will be undef forever.
If the caller is `y2_module_basetest->with_yast_env_variables()` it is the same as `with_yast_env_variables(y2_module_basetest)` 

So if you insist on 2 argument call, you need to use `@_` instead of `shift`. Sorry for the confusion, I was considering that you will drop the `y2_module_basetest` from the caller as well.

Reply:
done